server/aws-lsp-codewhisperer/src/language-server/agenticChat/qAgenticChatServer.ts

```diff
@@ -62,7 +65,7 @@ export const QAgenticChatServer =
 
         lsp.onInitialized(async () => {
             // Get initialized service manager and inject it to chatSessionManagementService to pass it down
-            amazonQServiceManager = AmazonQTokenServiceManager.getInstance()
```

Comment:
this auth manager should be decided at runtime, depending on user's auth mechanism

Reply:
When user change their settings, we already emit a message "onDidChangeConfiguration", you can re-use this to toggle the AUTH from IAM to bearer token.. 
```
lsp.onDidChangeConfiguration (

  if (IAM-auth) {
  
       amazonQServiceManager = AmazonQIAMServiceManager.getInstance()
  }


) 
```